docs/apigen.py

```diff
@@ -0,0 +1,255 @@
+# -*- coding: utf-8 -*-
+"""
+    apigen
+    ~~~~~~
+
+    This is a hacked up and heavily modified copy of sphinx.apidoc, which is
+    copyright the Sphinx team.
```

Comment:
Can we add a note about why this is necessary and `apidoc` won't do?

Reply:
Done.